### ISSUE TITLE
Build failure

### ISSUE THREAD

**Issue by haampie** [besser82/libxcrypt#162]
libxcrypt: 4.4.30
gcc: 11.2.0

```
lib/alg-yescrypt-opt.c:517:40: error: conversion to 'long long unsigned int' from 'long long int' may change the sign of the result [-Werror=sign-conversion]
```

Edit: I noticed #161, just not released yet.

**Comment by besser82** [comment]
@haampie, does applying the patch from #161, or https://github.com/besser82/libxcrypt/commit/17cf4ce8af5a2a1af4c0f52260019cbae01835b8, fix your issue?

**Comment by haampie** [comment]
Yes, fixes it. Maybe keep this issue open until the next release? Cause I looked first in open issues where I found nothing

**Comment by besser82** [comment]
Fixed in release [v4.4.31](https://github.com/besser82/libxcrypt/releases/tag/v4.4.31).
